src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,499 @@
+        ), [], false, true);
+
+        /*
+         * Enqueue all styles from assets URL map
+         */
+        foreach ($c->get('wp_bookings_ui/assets/styles') as $styleId => $styleDependency) {
+            wp_enqueue_style('rc-app-' . $styleId, $assetsUrlMap->get($styleDependency));
+        }
+
+        $state = $this->_onBookingsPage() ? $this->_getBookingsAppState($c) : $this->_getServiceAppState();
+
+        wp_localize_script('rc-app', 'EDDBK_APP_STATE', $state);
+    }
+
+    /**
+     * Register hook on admin init which will register everything else.
+     *
+     * @param EventManagerInterface $eventManager
+     * @param TemplateManager $templateManager
+     */
+    protected function _adminInit($eventManager, $templateManager, $c)
+    {
+        $metaboxConfig = $c->get('wp_bookings_ui/metabox');
+        /*
+         * Add metabox with availabilities configuration to
+         * service's edit page.
+         */
+        add_meta_box(
+            $metaboxConfig->get('id'),
+            $this->__($metaboxConfig->get('title')),
+            function () use ($templateManager) {
+                echo $this->_renderMetabox($templateManager);
+            },
+            $metaboxConfig->get('post_type')
+        );
+
+        /*
+         * Add screen options on bookings management page.
+         */
+        add_filter('screen_settings', function ($settings, \WP_Screen $screen) use ($templateManager) {
+            if (!$this->_onBookingsPage())
+                return $settings;
+
+            return $this->_renderBookingsScreenOptions($templateManager);
+        }, 10, 2);
+
+        /*
+         * @todo: this is not working as expected (nothing happens).
+         */
+//        $this->_attach('screen_settings', function ($event) use ($templateManager) {
+//            if (!$this->_onBookingsPage())
+//                return $event->getParam(0);
+//
+//            return $this->_renderBookingsScreenOptions($templateManager);
+//        }, 10);
+    }
+
+    /**
+     * Register pages in the admin menu.
+     *
+     * @param TemplateManager $templateManager
+     * @param ContainerInterface $c
+     */
+    protected function _adminMenu($templateManager, $c)
+    {
+        $rootMenuConfig = $c->get('wp_bookings_ui/menu/root');
+        $settingsMenuConfig = $c->get('wp_bookings_ui/menu/settings');
+        $aboutMenuConfig = $c->get('wp_bookings_ui/menu/about');
+
+        $this->bookingsPageId = add_menu_page(
+            $this->__($rootMenuConfig->get('page_title')),
+            $this->__($rootMenuConfig->get('menu_title')),
+            $rootMenuConfig->get('capability'),
+            $rootMenuConfig->get('menu_slug'),
+            function () use ($templateManager) {
+                echo $this->_renderMainPage($templateManager);
+            },
+            $rootMenuConfig->get('icon'),
+            $rootMenuConfig->get('position')
+        );
+
+        add_submenu_page(

```

Comment:
All this stuff can be achieved by using [rebelcode/wp-admin-menu][] and [rebelcode/wp-admin-page]. It will also allow you to move the definitions of the menus and pages into service definitions.

[rebelcode/wp-admin-menu]: https://github.com/RebelCode/wp-admin-menu
[rebelcode/wp-admin-page]: https://github.com/RebelCode/wp-admin-page